galaxy_ng/tests/integration/api/test_rbac_roles.py

```diff
@@ -0,0 +1,316 @@
+        # push_containers_to_existing_container_namespace,
+        # change_container_namespace,
+        # tag_untag_container_namespace,
+        # sync_remote_container,
+        create_exec_env,
+        create_container_registry_remote,
+        change_container_registry_remote,
+        delete_container_registry_remote,
+        # create_remote_container,
+        index_exec_env,
+    },
+    "galaxy.execution_environment_publisher": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        change_exec_env_desc,
+        change_exec_env_readme,
+        # create_containers_under_existing_container_namespace,
+        # push_containers_to_existing_container_namespace,
+        # change_container_namespace,
+        # tag_untag_container_namespace,
+        # sync_remote_container,
+        create_exec_env,
+    },
+    "galaxy.execution_environment_namespace_owner": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        # change_exec_env_desc_object,
+        # change_exec_env_readme_object,
+        # create_containers_under_existing_container_namespace_object,
+        # push_containers_to_existing_container_namespace_object,
+        # change_container_namespace_object,
+        # tag_untag_container_namespace_object,
+        # sync_remote_container_object,
+    },
+    "galaxy.execution_environment_collaborator": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        # change_exec_env_desc_object,
+        # change_exec_env_readme_object,
+        # push_containers_to_existing_container_namespace_object,
+        # tag_untag_container_namespace_object,
+        # sync_remote_container_object,
+    },
+    "galaxy.group_admin": {
+        upload_collection_to_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        add_groups,
+        change_groups,
+        delete_groups,
+    },
+    "galaxy.user_admin": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        view_users,
+        delete_users,
+        add_users,
+        change_users,
+    },
+    "galaxy.task_admin": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_groups,
+        view_tasks,
+    }
+}
+
+
+@pytest.mark.role_rbac
+@pytest.mark.parametrize("role", ROLES_TO_TEST)
+def test_role_actions(role):
```

Comment:
Some thoughts on optimization:

We're wasting a ton of time in the tests creating objects for actions that need to happen on specific objects (change, delete, approve, upload_to etc). I'm thinking it might be best to initialize each object for these test once per role and then pass the object PK or ref to the action. The only caveat is that the delete tests would need to come last.